### DISCUSSION TITLE
Input crystal orientation into tess file

### DISCUSSION THREAD

**Discussion by lreigbua** [neperfepx/neper#510] [Q&A] [answered]
Hello,

I have a tess file for which I have calculated the orientations externally. I would like to input these orientations into the tess file to visualize an IPF.

I have loaded the tess file generated and tried to add the orientations through an ori file but it keeps the original orientations:

```
neper -T -n "from_morpho" -morpho "file(Polycrystal_laminated_3scale.tess)" -ori "msfile(ori,des=euler-bunge)" -format tess,ori -oridescriptor euler-bunge -statcell "scaleid(1)" -o test
neper -V test.tess -datacellcol ori -print img
```

I saw in other discussions that you can do it through ori file when creating the tesselation initially, but this is very tedious to code for determinate multi-scale tessellations and requires to rerun the tesselation to know the number of laths. Is there no way to do it by loading the tess file like I did?

Thanks a lot!

**Comment by rquey** [comment, marked as answer]
I've just added `-transform ori` for tesses:

> `ori(<file_name>[,des=<descriptor>])`: override cell orientations with ones defined in a [Data File](https://neper.info/doc/fileformat.html#data-file) written using a specific descriptor (see [Rotations and Orientations](https://neper.info/doc/exprskeys.html#rotations-and-orientations), default `rodrigues`).

It works equally well for single and multi-scale tessellations.

So, once you have your tessellation, use something ike `neper -T -loadtess my_tess.tess -transform "ori(...)" -o my_tess2`.

Needs version 4.4.2-31.

**Comment by lreigbua** [comment]
Awesome! Thanks Romain
